cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/SparkCqlEvaluator.java

```diff
@@ -730,28 +771,16 @@ protected CqlEvaluationRequests readJobSpecification(String path) throws Excepti
      * @param outputURI        URI pointing at the location where output data should
      *                         be written.
      */
-    protected void writeResults(SparkSession spark, StructType schema, JavaPairRDD<Object, Map<String, Object>> resultsByContext,
+    protected void writeResults(SparkSession spark, StructType schema, JavaPairRDD<Object, Row> resultsByContext,
             String outputURI) {
         Dataset<Row> dataFrame = spark.createDataFrame(
-                resultsByContext
-                        .map(t -> {
-                            Object contextKey = t._1();
-                            Map<String, Object> results = t._2();
-
-                            Object[] data = new Object[schema.fields().length];
-                            data[0] = contextKey;
-                            for (int i = 1; i < schema.fieldNames().length; i++) {
-                                data[i] = results.get(schema.fieldNames()[i]);
-                            }
-                            return data;
-                        })
-                        .map(RowFactory::create),
+                resultsByContext.values(),
                 schema
         );
 
-		if (args.outputPartitions != null) {
-			dataFrame = dataFrame.repartition(args.outputPartitions);
-		}
+        if (args.outputPartitions != null) {
+        	dataFrame = dataFrame.repartition(args.outputPartitions);
```

Comment:
I think the 3rd leading tab is still here.

Reply:
Will fix.